### PR TITLE
Cache mvn dependencies for cron workflow ITs

### PR DESCRIPTION
**Pull request by tejaswini-imply**
- With this PR, mvn dependencies are cached prior to running jdk11 ITs in cron workflow.
- Only security-vulnerabilities job in this workflow runs on PR if there are changes to `owasp-dependency-check-suppressions.xml` file.

### CODE REVIEW

**Review by abhishekagarwal87 — APPROVED** [review]

File: .github/workflows/cron-job-its.yml [hunk 1]
```diff
@@ -17,14 +17,44 @@ name: Cron Job ITs
...
+        id: maven
+        uses: actions/cache@v3
+        with:
+          path: ~/.m2/repository
+          key: maven-${{ runner.os }}-8-${{ github.sha }}
+
+      - name: Maven build
+        id: maven_build
+        run: |
+          ./it.sh ci
```
Comment (abhishekagarwal87) [inline .github/workflows/cron-job-its.yml:49]: this works for old IT workflow as well?

Comment (tejaswini-imply) [reply .github/workflows/cron-job-its.yml:49]: Yes, build happens the same way independently in the old (standard&revised) IT workflow. Also, old IT workflow builds docker container for revised ITs. Its not needed here as of now.

File: .github/workflows/cron-job-its.yml [hunk 2]
```diff
@@ -82,8 +115,16 @@ jobs:
       - name: Checkout branch
         uses: actions/checkout@v3
 
+      - name: setup java
```
Comment (imply-elliott) [inline .github/workflows/cron-job-its.yml:118]: Just a note, we're using the on-runner java now to save time.
Example here: https://github.com/implydata/druid/blob/c882bd95ceea3df9d5dc3624cef6a25681f1af39/.github/workflows/unit-and-integration-tests-unified.yml#L63

Comment (tejaswini-imply) [reply .github/workflows/cron-job-its.yml:118]: I understand that, I have used this setup only for `security_vulnerabilities` job to take the advantage of GHA powered maven cache and incase of CVE suppression PRs `security_vulnerabilities` job alone runs and still cache maven dependencies across workflows.

Comment (imply-elliott) [reply .github/workflows/cron-job-its.yml:118]: That makes sense. However, we can do the same by using the same caching parameters the setup java action uses, which just follows this convention: https://github.com/actions/cache/blob/0638051e9af2c23d10bb70fa9beffcad6cff9ce3/examples.md#java---maven
Again, not a big deal, but will save a bit of time downloading and setting up java.

**Pull request merged**